metadata/metadata.go

```diff
@@ -622,6 +649,30 @@ func (resource *featureVariantResource) Update(lookup ResourceLookup, updateRes
 	return nil
 }
 
+func (resource *featureVariantResource) IsEquivalent(other ResourceVariant) bool {
+	otherVariant, ok := other.(*featureVariantResource)
+	if !ok {
+		return false
+	}
+
+	thisProto := resource.serialized
+	otherProto := otherVariant.serialized
+
+	if thisProto.GetName() == otherProto.GetName() &&
+		proto.Equal(thisProto.GetSource(), otherProto.GetSource()) &&
+		(proto.Equal(thisProto.GetFunction(), otherProto.GetFunction()) ||
+			proto.Equal(thisProto.GetColumns(), otherProto.GetColumns()) &&
+				thisProto.GetProvider() == otherProto.GetProvider()) {
+
+		return true
+	}
```

Comment:
Given this boolean logic, it would be nice to have a comment that explains the ways two feature variants could be considered equivalent. That way, if someone else come to this to extend it, it'll be easier to reason about the requirements IMHO.

Reply:
yeahhh agreed. I was even considering having a set of fields on Resource Variants that dictate the set of equivalent fields